CorsixTH/Lua/world.lua

```diff
@@ -1129,15 +1112,11 @@ function World:onEndDay()
   end
 
   -- check if it's time for an earthquake, and the user is at least on level 5
-  if (self.year - 1) * 12 + self.month == self.next_earthquake_month and
-      self.day == self.next_earthquake_day then
+  if self.next_earthquake and self.next_earthquake.size and
+      (self.year - 1) * 12 + self.month == self.next_earthquake.start_month and
```

Comment:
Probably not worth doing for this PR, but `(self.year - 1) * 12 + self.month` almost seems worth putting in a function at some point.